### ISSUE TITLE
Upgrading to 3.1.19 breaks validation rule "image"

### ISSUE THREAD

**Issue by grischaerbe** [statamic/cms#3811]
## Bug Description

After upgrading an existing site to 3.1.19 it is not possible to save entries with fields of type "assets"  which have a validation rule "image" applied. The Error that gets printed out as a toast is: `Error: Argument #1 ($haystack) must be of type string, Statamic\Fieldtypes\Assets\ImageRule given`

## How to Reproduce

Add an assets field to a blueprint, add the validation rule "image" to it, make an entry and try to save it, get an error.

## Environment

Statamic 3.1.19 Pro
Laravel 8.38.0
PHP 8.0.3
aryehraber/statamic-logbook 2.0.2
edalzell/forma 0.8.12
legrisch/statamic-graphql-thumbnails 0.3.0
legrisch/statamic-webhooks 0.3.4
simonhamp/gitamic 1.7.4

**Statamic version:** 3.1.19

**PHP version:** 8.0.3

**Install method:**
- install from `statamic/statamic`

**Comment by grischaerbe** [comment]
An existing and previously working validation rule `mimetypes:audio/mpeg,audio/mp4` on an asset field is also failing: `Error: Argument #1 ($haystack) must be of type string, Statamic\Fieldtypes\Assets\MimetypesRule given`.

**Comment by jasonvarga** [comment]
Which version did you upgrade _from_ ?

**Comment by vluijkx** [comment]
We have got the same issue upgrading from 3.1.18.

**Comment by jasonvarga** [comment]
Can you provide the blueprint/fieldset yaml file that holds the assets field?

**Comment by simonolog** [comment]
Having the same issue. Here's the blueprint of (one of) the affected pages:

```yaml
title: Person
sections:
  main:
    display: Main
    fields:
      -
        handle: title
        field:
          type: text
          required: true
          validate:
            - required
          display: Name
          listable: hidden
          input_type: text
          antlers: false
      -
        handle: photo
        field:
          mode: grid
          restrict: true
          allow_uploads: true
          max_files: 1
          display: Portraitfoto
          type: assets
          icon: assets
          listable: hidden
          validate:
            - required
            - image
          container: assets
          folder: personen
      -
        import: wysiwyg_default
      -
        [some more simple text fields]
  sidebar:
    display: Sidebar
    fields:
      -
        handle: slug
        field:
          type: slug
          required: true
          localizable: true
          validate:
            - required
            - 'unique_entry_value:{collection},{id},{site}'
      -
        handle: topic
        field:
          type: terms
          taxonomies:
            - topic
          display: Thema
          mode: select
```

On removing the `image` validation, everything works fine.

**Comment by jasonvarga** [comment]
Fixed for 3.1.21
